hot-reload-agent/src/main/kotlin/org/jetbrains/compose/reload/agent/recompiler.kt

```diff
@@ -35,34 +41,61 @@ private val recompileRequests = LinkedBlockingQueue<RecompileRequest>(
 )
 
 internal fun launchRecompiler() {
-    val gradleBuildRoot = gradleBuildRoot ?: run {
-        logger.error("Missing 'compose.build.root' property")
-        return
-    }
-
-    val gradleBuildProject = gradleBuildProject ?: run {
-        logger.error("Missing 'compose.build.project' property")
-        return
-    }
-
-    val gradleBuildTask = gradleBuildTask ?: run {
-        logger.error("Missing 'compose.build.compile.task' property")
-        return
-    }
 
     val port = ComposeHotReloadAgent.orchestration.port
     logger.debug("'Compose Recompiler': Using orchestration at '$port'")
 
+    val processBuilder = when (buildSystem ?: "gradle") {
+        "amper" -> {
+            val amperBuildRoot = amperBuildRoot ?: run {
+                logger.error("Missing '${HotReloadProperty.AmperBuildRoot.key}' property")
+                return
+            }
+            
+            val amperBuildTask = amperBuildTask ?: run {
+                logger.error("Missing '${HotReloadProperty.AmperBuildTask.key}' property")
+                return
+            }
+
+            createRecompilerProcessBuilder(
+                amperBuildRoot = amperBuildRoot,
+                amperBuildTask = amperBuildTask,
+                orchestrationPort = port
+            )
+        }
+        "gradle" -> {
+            val composeBuildRoot = gradleBuildRoot ?: run {
+                logger.error("Missing '${HotReloadProperty.GradleBuildRoot.key}' property")
+                return
+            }
+
+            val gradleBuildProject = gradleBuildProject ?: run {
+                logger.error("Missing '${HotReloadProperty.GradleBuildProject.key}' property")
+                return
+            }
+
+            val gradleBuildTask = gradleBuildTask ?: run {
+                logger.error("Missing '${HotReloadProperty.GradleBuildTask.key}' property")
+                return
+            }
+
+            createRecompilerProcessBuilder(
+                gradleBuildRoot = composeBuildRoot,
+                gradleBuildProject = gradleBuildProject,
+                gradleBuildTask = gradleBuildTask,
+                orchestrationPort = port
+            )
+        }
+
+        else -> {
+            logger.error("${HotReloadProperty.BuildSystem.key} has incorrect value: $buildSystem")
```

Comment:
Maybe use logger.orchestration here, to send this to other tools, as this would be a very important message which should also be displayed in the IDE or dev tooling!